tekton/ci/repos/pipeline/template.yaml

```diff
@@ -51,3 +51,53 @@
           value: test/e2e-tests.sh
         - name: e2e-env
           value: test/e2e-tests-kind.env
+  - apiVersion: tekton.dev/v1beta1
+    kind: PipelineRun
+    metadata:
+      generateName: run-go-coverage-
+      labels:
+        prow.k8s.io/build-id: $(tt.params.buildUUID)
+        tekton.dev/source-event-id: $(tt.params.sourceEventId)
+        tekton.dev/check-name: pull-tekton-pipeline-go-coverage
```

Comment:
The name is already taken by prow, so we should use a slightly different name here

Reply:
and I've just done a quick check, and I'm pretty sure `tekton.dev/check-name` is the value that is used for the GitHub check name by the existing automation.

Reply:
I changed it to `pull-tekton-pipeline-go-coverage-using-tekton` to indicate that the job was run using the Tekton dogfooding cluster. I'm happy to change it to something else too.